src/sympc/store/crypto_primitive_provider.py

```diff
@@ -43,9 +46,18 @@ def generate_primitives(
             raise ValueError(f"{op_str} not registered")
 
         generator = CryptoPrimitiveProvider._func_providers[op_str]
-
         primitives = generator(**g_kwargs)
 
+        if CryptoPrimitiveProvider._logging and (
+            p_kwargs.get("a_shape") and p_kwargs.get("b_shape")
```

Comment:
Q: Can't we always presume we have ```a_shape``` and ```b_shape```?

Reply:
I wasnt sure about it... do we have any other way/idea ?

Reply:
If we simply remove them, do they fail?

Reply:
Discussed on slack :D